src/parser/node.js

```diff
@@ -1,13 +1,16 @@
+// @flow
+
 import Parser from "./index";
 import UtilParser from "./util";
 import { SourceLocation, type Position } from "../util/location";
+import type { Comment, Node as NodeType, NodeBase } from "../types";
 
 // Start an AST node, attaching a start offset.
 
 const commentKeys = ["leadingComments", "trailingComments", "innerComments"];
 
-class Node {
-  constructor(parser?: Parser, pos?: number, loc?: Position) {
+class Node implements NodeBase {
+  constructor(parser: Parser, pos: number, loc: Position) {
```

Comment:
This constructor was only called with missing parameters in `__clone`, so better to declare them as required.